src/main/java/com/jpa/services/DeveloperService.java

```diff
@@ -0,0 +1,40 @@
+package com.jpa.services;
+
+import com.jpa.entity.Developer;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+
+public class DeveloperService implements DAO<Developer>{
+    private EntityManager entityManager = Persistence.createEntityManagerFactory("JPA").createEntityManager();
+
+    public Developer read(int id) {
+        entityManager.getTransaction().begin();
+        Developer developer = entityManager.find(Developer.class, id);
+        entityManager.getTransaction().commit();
+        entityManager.close();
```

Comment:
In case of exception situation should we do rollback?
The same about other methods.